gremlin-test/src/main/resources/org/apache/tinkerpop/gremlin/test/features/filter/All.feature

```diff
@@ -0,0 +1,152 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+@StepClassFilter @StepAll
+Feature: Step - all()
+
+  Scenario: g_V_valuesXageX_allXgtX32XX
+    Given the modern graph
+    And the traversal of
+      """
+      g.V().values("age").all(P.gt(32))
+      """
+    When iterated to list
+    Then the result should be empty
+
+  Scenario: g_V_valuesXageX_whereXisXP_gtX33XXX_fold_allXgtX33XX
+    Given the modern graph
+    And the traversal of
+      """
+      g.V().values("age").where(__.is(P.gt(33))).fold().all(P.gt(33))
+      """
+    When iterated to list
+    Then the result should be unordered
+      | result |
+      | l[d[35].i] |
+
+  Scenario: g_V_valuesXageX_order_byXdescX_fold_allXgtX10XX
+    Given the modern graph
+    And the traversal of
+      """
+      g.V().values("age").order().by(desc).fold().all(P.gt(10))
+      """
+    When iterated to list
+    Then the result should be unordered
+      | result |
+      | l[d[35].i,d[32].i,d[29].i,d[27].i] |
+
+  Scenario: g_V_valuesXageX_order_byXdescX_fold_allXgtX30XX
+    Given the modern graph
+    And the traversal of
+      """
+      g.V().values("age").order().by(desc).fold().all(P.gt(30))
+      """
+    When iterated to list
+    Then the result should be empty
+
+  @GraphComputerVerificationInjectionNotSupported
+  Scenario: g_injectXabc_bcdX_allXeqXbcdXX
+    Given the empty graph
+    And using the parameter xx1 defined as "l[abc,bcd]"
+    And the traversal of
+      """
+      g.inject(xx1).all(P.eq("bcd"))
+      """
+    When iterated to list
+    Then the result should be empty
+
+  @GraphComputerVerificationInjectionNotSupported
+  Scenario: g_injectXbcd_bcdX_allXeqXbcdXX
+    Given the empty graph
+    And using the parameter xx1 defined as "l[bcd,bcd]"
+    And the traversal of
+      """
+      g.inject(xx1).all(P.eq("bcd"))
+      """
+    When iterated to list
+    Then the result should be unordered
+      | result |
+      | l[bcd,bcd] |
+
+  @GraphComputerVerificationInjectionNotSupported
+  Scenario: g_injectXnull_abcX_allXTextP_startingWithXaXX
+    Given the empty graph
+    And using the parameter xx1 defined as "l[null,abc]"
```

Comment:
interesting the parsers for `l[]` work like this. i guess it is treating "null" as `null` and not as a `String` and treating "abc" without the double quotes as a `String` somehow?